### PR TITLE
od: simplify od module to only provide od_hex_dump

### PR DESCRIPTION
**Pull request by miri64**
To my knowledge no one was using the more complex functionalities of this module (except for the test application and pktbuf which just used the same flags as the `od_hex_dump()` wrapper), so why not reduce the functionality of this module?

Depends on #6551.

### CODE REVIEW

**Comment by miri64** [comment]
Of course this frees some space in the `gnrc_networking` application:

```
-1972	0	0	-1972	arduino-due
61556	664	17600	79820	old
59584	664	17600	77848	new

-1450	-224	0	-1674	arduino-duemilanove
72466	12662	11408	96536	old
71016	12438	11408	94862	new

-1516	-224	0	-1740	arduino-mega2560
74306	12682	11424	98412	old
72790	12458	11424	96672	new

-1450	-224	0	-1674	arduino-uno
72466	12662	11408	96536	old
71016	12438	11408	94862	new

-2116	0	0	-2116	arduino-zero
64076	600	17660	82336	old
61960	600	17660	80220	new

-3060	0	0	-3060	avsextrem
89836	596	97708	188140	old
86776	596	97708	185080	new

-1984	0	0	-1984	cc2538dk
78376	600	20872	99848	old
76392	600	20872	97864	new

-1972	0	0	-1972	cc2650stk
61068	600	17780	79448	old
59096	600	17780	77476	new

-1976	0	0	-1976	ek-lm4f120xl
61808	600	17524	79932	old
59832	600	17524	77956	new

-1960	0	0	-1960	f4vi1
61732	600	17644	79976	old
59772	600	17644	78016	new

-1988	0	0	-1988	fox
78612	600	20760	99972	old
76624	600	20760	97984	new

-1972	0	0	-1972	frdm-k64f
62504	1628	18088	82220	old
60532	1628	18088	80248	new

-1988	0	0	-1988	iotlab-a8-m3
78628	600	20760	99988	old
76640	600	20760	98000	new

-1988	0	0	-1988	iotlab-m3
78628	600	20760	99988	old
76640	600	20760	98000	new

-1972	0	0	-1972	limifrog-v1
61688	600	17652	79940	old
59716	600	17652	77968	new

-1972	0	0	-1972	maple-mini
61908	600	17676	80184	old
59936	600	17676	78212	new

-1968	0	0	-1968	mbed_lpc1768
61164	600	17516	79280	old
59196	600	17516	77312	new

-3084	0	0	-3084	msba2
113032	596	97708	211336	old
109948	596	97708	208252	new

-1972	0	0	-1972	msbiot
62176	600	17668	80444	old
60204	600	17668	78472	new

-1984	0	0	-1984	mulle
81220	1668	21256	104144	old
79236	1668	21256	102160	new

-2918	-4	0	-2922	native
193045	600	104884	298529	old
190127	596	104884	295607	new

-2116	0	0	-2116	nucleo-f091
63456	600	17660	81716	old
61340	600	17660	79600	new

-1972	0	0	-1972	nucleo-f207
62036	600	17676	80312	old
60064	600	17676	78340	new

-1972	0	0	-1972	nucleo-f303
61980	600	17660	80240	old
60008	600	17660	78268	new

-1972	0	0	-1972	nucleo-f401
61992	600	17652	80244	old
60020	600	17652	78272	new

-1956	0	0	-1956	nucleo-f446
62012	600	17660	80272	old
60056	600	17660	78316	new

-1972	0	0	-1972	nucleo-l1
61500	600	17644	79744	old
59528	600	17644	77772	new

-1972	0	0	-1972	opencm9-04
61688	600	17668	79956	old
59716	600	17668	77984	new

-1984	0	0	-1984	openmote-cc2538
78376	600	20872	99848	old
76392	600	20872	97864	new

-1988	0	0	-1988	pba-d-01-kw2x
79780	1628	21268	102676	old
77792	1628	21268	100688	new

-1984	0	0	-1984	remote-pa
79052	600	20872	100524	old
77068	600	20872	98540	new

-1984	0	0	-1984	remote-reva
79032	600	20872	100504	old
77048	600	20872	98520	new

-1984	0	0	-1984	remote-revb
79040	600	20872	100512	old
77056	600	20872	98528	new

-2116	0	0	-2116	saml21-xpro
63452	600	17644	81696	old
61336	600	17644	79580	new

-2124	0	0	-2124	samr21-xpro
81216	600	20760	102576	old
79092	600	20760	100452	new

-1968	0	0	-1968	seeeduino_arch-pro
61164	600	17516	79280	old
59196	600	17516	77312	new

-1972	0	0	-1972	slwstk6220a
61244	600	17644	79488	old
59272	600	17644	77516	new

-2116	0	0	-2116	sodaq-autonomo
64128	600	17676	82404	old
62012	600	17676	80288	new

-1972	0	0	-1972	stm32f3discovery
62060	600	17660	80320	old
60088	600	17660	78348	new

-1972	0	0	-1972	stm32f4discovery
62188	600	17660	80448	old
60216	600	17660	78476	new

-1972	0	0	-1972	udoo
61556	664	17600	79820	old
59584	664	17600	77848	new

-1476	-224	0	-1700	waspmote-pro
73772	12678	11416	97866	old
72296	12454	11416	96166	new
```

**Comment by miri64** [comment]
Can somebody also please now check the test script with a non-native platform

**Comment by kaspar030** [comment]
Did you consider using fmt? Seems like at least some functionality is duplicated.

**Comment by miri64** [comment]
@kaspar030 you mean instead of printf? Will do.

**Comment by miri64** [comment]
@kaspar030 `fmt` does not support padding with `0`s right?

**Comment by kaspar030** [comment]
> @kaspar030 fmt does not support padding with 0s right?

Nope, but shouldn't that be a five-liner? But I guess we can consolidate later. ;)

**Comment by miri64** [comment]
Still not sure if you mean I should *use* `fmt` or if I should *include* `od_hex_dump()` into `fmt`. I'm against the latter. `fmt` is for _formatting_, not memory dumping. That's what `od` is and was always for. That's why it is used all through-out the code-base for that (`MODULE_OD` is even used as a switch for dumping or not dumping).

**Comment by miri64** [comment]
(I've seen `fmt_bytes_hex_reverse()` and I'm not sure what it does and if you are referring to it *if* you want me to merge `od` and `fmt`. If it indeed is for dumping, I would rather move it to `od`).

**Comment by A-Paul** [comment]
Is my assumption correct that *od* (octal dump) can't dump in octal format anymore? :D

**Comment by miri64** [comment]
Yes :-D

**Comment by A-Paul** [comment]
Shouldn't it be renamed to *hd* then?

**Comment by miri64** [comment]
Why? It still dumps objects not hexes (and also I like to avoid another rename litany all throughout the code base ;-))

**Comment by miri64** [comment]
Since I ported it to use `fmt`, this PR is waiting for #6551 now.

**Comment by miri64** [comment]
Rebased to current master, since #6551 got merged.

**Comment by miri64** [comment]
@kaspar030 @A-Paul ping?

**Comment by kaspar030** [comment]
The changes look OK, but I don't have time to test. Possibly later today or tomorrow.

**Comment by A-Paul** [comment]
Run *tests/od* with `make test` on *gcc (Debian 4.9.2-10) 4.9.2* was successful.

I still don't like the idea of throwing things away without really knowing who is using them right now.

**Comment by miri64** [comment]
Even if: this feature was always only thought to be used for debugging, so people will adapt.

**Comment by miri64** [comment]
Rebased

**Comment by miri64** [comment]
Can we get this merged today?

File: sys/include/od.h [hunk 1]
```diff
@@ -34,129 +27,20 @@ extern "C" {
 #include <stdint.h>
```
Comment (OlegHahm) [inline sys/include/od.h]: `stddef.h` is required, too (for `size_t`).

Comment (miri64) [inline sys/include/od.h]: Fixed

**Comment by miri64** [comment]
Squashed

**Comment by miri64** [comment]
Fixed Doc error reported by Murdock2 and squashed immediately

**Pull request merged**